providers/byteplus/byteplusProvider.go

```diff
@@ -0,0 +1,347 @@
+	ak, sk := m["ak"], m["sk"]
+
+	client := byteplus.InitDNSBytePlusClient()
+	client.SetAccessKey(ak)
+	client.SetSecretKey(sk)
+
+	provider := byteplusProvider{client: client}
+
+	return &provider, nil
+}
+
+var features = providers.DocumentationNotes{
+	// The default for unlisted capabilities is 'Cannot'.
+	// See providers/capabilities.go for the entire list of capabilities.
+	providers.CanGetZones:            providers.Can(),
+	providers.CanConcur:              providers.Cannot(),
+	providers.CanUseAlias:            providers.Cannot(),
+	providers.CanUseCAA:              providers.Can(),
+	providers.CanUseLOC:              providers.Cannot(),
+	providers.CanUsePTR:              providers.Cannot(),
+	providers.CanUseSRV:              providers.Can(),
+	providers.CanUseTLSA:             providers.Cannot(),
+	providers.DocCreateDomains:       providers.Cannot(),
+	providers.DocDualHost:            providers.Cannot(),
+	providers.DocOfficiallySupported: providers.Cannot(),
+}
+
+func init() {
+	fns := providers.DspFuncs{
+		Initializer:   NewByteplus,
+		RecordAuditor: AuditRecords,
+	}
+	providers.RegisterDomainServiceProviderType("BYTEPLUS", fns, features)
+}
+
+// EnsureZoneExists creates a zone if it does not exist
+func (c *byteplusProvider) EnsureZoneExists(domain string) error {
+	_, err := c.findDomainByName(domain)
+
+	return err
+}
+
+// GetNameservers returns the nameservers for a domain.
+func (c *byteplusProvider) GetNameservers(domain string) ([]*models.Nameserver, error) {
+	return models.ToNameservers(defaultNS)
+}
+
+// GetZoneRecords gets the records of a zone and returns them in RecordConfig format.
+func (c *byteplusProvider) GetZoneRecords(domainName string, meta map[string]string) (models.Records, error) {
+	//dc.Punycode()
+
+	domain, err := c.findDomainByName(domainName)
+	if err != nil {
+		return nil, err
+	}
+	domainID := *domain.ZID
+
+	// warning!! inconsistency within byteplus go sdk.
+	// ListRecords later below is demanding string version if ZID (domain id)
+	// while TopZoneResponse (findDomainByName) above returns in int64.
+	domainIDString := strconv.Itoa(int(domainID))
+
+	// Create the struct, setting the pointer to the string
+	domainStr := domainStruct{
+		DomainID: &domainIDString, // Take the address of domainIDStr
+	}
+
+	ctx := context.Background()
+	pageSize := "500" // arbitrary limit. max records size per page is 500.
+	records, err := c.client.ListRecords(ctx, &byteplus.ListRecordsRequest{ZID: domainStr.DomainID, PageSize: &pageSize})
+	if err != nil {
+		return nil, err
+	}
+
+	existingRecords := make([]*models.RecordConfig, 0, len(records.Records))
+	for _, r := range records.Records {
+		if r.RecordID == nil {
+			continue
+		}
+
+		recordID := *r.RecordID
+
+		record, err := c.client.QueryRecord(ctx, &byteplus.QueryRecordRequest{RecordID: &recordID})
+		if err != nil {
+			return nil, err
+		}
+
+		// nil pointers are not expected, but just to be on the safe side...
```

Comment:
Suggestion: Split this out to a function called nativeToRC() which takes `record` and returns a `*models.RecordConfig()`.  This will improve readability and testability.